flytepropeller/pkg/controller/nodes/attr_path_resolver.go

```diff
@@ -79,13 +88,93 @@ func resolveAttrPathInPbStruct(nodeID string, st *structpb.Struct, bindAttrPath
 	}
 
 	// After resolve, convert the interface to literal
-	literal, err := convertInterfaceToLiteral(nodeID, currVal)
+	literal, err := convertStructToLiteral(nodeID, currVal)
 
 	return literal, err
 }
 
-// convertInterfaceToLiteral converts the protobuf struct (e.g. dataclass) to literal
-func convertInterfaceToLiteral(nodeID string, obj interface{}) (*core.Literal, error) {
+// resolveAttrPathInJSON resolves the msgpack bytes (e.g. dataclass) with attribute path
+func resolveAttrPathInJSON(nodeID string, msgpackBytes []byte, bindAttrPath []*core.PromiseAttribute) (*core.Literal,
+	error) {
+
+	var currVal interface{}
+	var tmpVal interface{}
+	var exist bool
+	var jsonStr string
+
+	err := msgpack.Unmarshal(msgpackBytes, &jsonStr)
+	if err != nil {
+		return nil, err
+	}
+
+	// Golang has problem with unmarshalling integer as float64
+	// reference: https://stackoverflow.com/questions/22343083/json-unmarshaling-with-long-numbers-gives-floating-point-number
+
+	decoder := json.NewDecoder(strings.NewReader(jsonStr))
+	decoder.UseNumber()
+
+	err = decoder.Decode(&tmpVal)
+	if err != nil {
+		return nil, err
+	}
+	currVal = convertNumbers(tmpVal)
+
+	// Turn the current value to a map so it can be resolved more easily
```

Comment:
hmm, where did you turn the current value to a map?

Reply:
This is my fault, nice catch!!